src/js/components/Hero.js

```diff
@@ -0,0 +1,148 @@
+// (C) Copyright 2016 Hewlett Packard Enterprise Development LP
+
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import CSSClassnames from '../utils/CSSClassnames';
+import Box from 'grommet/components/Box';
+import Image from 'grommet/components/Image';
+import Video from 'grommet/components/Video';
+
+const CLASS_ROOT = CSSClassnames.HERO;
+const LIGHT_COLORINDEX = 'light-1';
+const DARK_COLORINDEX = 'grey-1';
+const PALM_BREAKPOINT = 720;
+
+export default class Hero extends Component {
+
+  constructor (props) {
+    super(props);
+    this._setReverse = this._setReverse.bind(this);
+    this._setBackgroundColorIndex = this._setBackgroundColorIndex.bind(this);
+
+    this.state = {
+      colorIndex: props.darkTheme ? DARK_COLORINDEX : LIGHT_COLORINDEX,
+      reverse: (props.justify === 'start') ? true : false
+    };
+  }
+
+  componentDidMount () {
+    window.addEventListener('resize', this._setReverse);
+    window.addEventListener('resize', this._setBackgroundColorIndex);
+    this._setReverse();
+    this._setBackgroundColorIndex();
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('resize', this._setReverse);
+    window.removeEventListener('resize', this._setBackgroundColorIndex);
+  }
+
+  _setBackgroundColorIndex () {
+    const { darkTheme } = this.props;
+
+    if (window.innerWidth < PALM_BREAKPOINT) {
+      this.setState({ colorIndex: LIGHT_COLORINDEX });
+    } else {
+      this.setState({ colorIndex: darkTheme ? DARK_COLORINDEX : LIGHT_COLORINDEX });
+    }
+  }
+
+  _setReverse () {
+    const { justify } = this.props;
+
+    if (window.innerWidth < PALM_BREAKPOINT) {
+      this.setState({ reverse: false });
+    } else {
+      this.setState({ reverse: (justify === 'start') ? true : false });
+    }
+  }
+
+  render () {
+    const { backgroundImage, backgroundType, backgroundVideoLoop, backgroundVideoMuted, backgroundVideoPoster, backgroundVideoSource, children, className, flush, image, justify, responsiveBackgroundPosition, separator, size } = this.props;
```

Comment:
Alan and I have been slowly changing files to have at most 80 characters per line, which makes reviewing them easier, among other things. This one seems a bit long.